packages/file_selector/file_selector_windows/lib/src/file_selector_dart/file_dialog_controller.dart

```diff
@@ -0,0 +1,91 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:ffi';
+
+import 'package:ffi/ffi.dart';
+import 'package:win32/win32.dart';
+
+import 'ifile_open_dialog_factory.dart';
+
+/// A thin wrapper for IFileDialog to allow for faking and inspection in tests.
+///
+/// Since this class defines the end of what can be unit tested, it should
+/// contain as little logic as possible.
+class FileDialogController {
+  /// Creates a controller managing [IFileDialog](https://pub.dev/documentation/win32/latest/winrt/IFileDialog-class.html).
+  /// It also receives an IFileOpenDialogFactory to construct [IFileOpenDialog]
+  /// instances.
+  FileDialogController(
+    IFileDialog fileDialog,
+    IFileOpenDialogFactory iFileOpenDialogFactory,
+  )   : _fileDialog = fileDialog,
+        _iFileOpenDialogFactory = iFileOpenDialogFactory;
+
+  /// The [IFileDialog] to work with.
+  final IFileDialog _fileDialog;
```

Comment:
I may be missing something, but what `free`s this?

Reply:
Good catch, done!